### PR TITLE
Add allowList for Duck Player message handlers

### PR DESCRIPTION
**Pull request by CrisBarreiro**
Task/Issue URL: https://app.asana.com/0/72649045549333/1208375755367840/f

### Description
Only process duck-player related messages if the origin belongs to a given set of domains

### Steps to test this PR

_Feature 1_
- [x] Open a YouTube video 
- [x] Click the button to open in Duck Player
- [x] Check the app navigates to Duck Player
- [x] Click the settings button
- [x] Check native settings are launched

### CODE REVIEW

**Comment by CrisBarreiro** [comment]
* **#5053** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/5053?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `develop`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @CrisBarreiro and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: content-scope-scripts/content-scope-scripts-impl/src/main/java/com/duckduckgo/contentscopescripts/impl/messaging/ContentScopeScriptsJsMessaging.kt [hunk 1]
```diff
@@ -46,6 +50,7 @@ class ContentScopeScriptsJsMessaging @Inject constructor(
     private val jsMessageHelper: JsMessageHelper,
     private val dispatcherProvider: DispatcherProvider,
     private val coreContentScopeScripts: CoreContentScopeScripts,
+    private val duckPlayer: DuckPlayer,
```
Comment (marcosholgado) [inline content-scope-scripts/content-scope-scripts-impl/src/main/java/com/duckduckgo/contentscopescripts/impl/messaging/ContentScopeScriptsJsMessaging.kt:53]: This is not used

File: duckplayer/duckplayer-api/src/main/java/com/duckduckgo/duckplayer/api/DuckPlayer.kt [hunk 2]
```diff
@@ -174,6 +177,13 @@ interface DuckPlayer {
      */
     suspend fun willNavigateToDuckPlayer(destinationUrl: Uri): Boolean
 
+    /**
+     * Retrieves the YouTube embed URL.
+     *
+     * @return The YouTube embed URL.
+     */
+    suspend fun getYouTubeEmbedUrl(): String
```
[the remark this answers is not in the document]
Comment (CrisBarreiro) [reply duckplayer/duckplayer-api/src/main/java/com/duckduckgo/duckplayer/api/DuckPlayer.kt]: Fixed

File: content-scope-scripts/content-scope-scripts-impl/src/main/java/com/duckduckgo/contentscopescripts/impl/messaging/ContentScopeScriptsJsMessaging.kt (hunk 1 above)
Comment (CrisBarreiro) [reply content-scope-scripts/content-scope-scripts-impl/src/main/java/com/duckduckgo/contentscopescripts/impl/messaging/ContentScopeScriptsJsMessaging.kt]: Fixed

File: duckplayer/duckplayer-impl/src/main/java/com/duckduckgo/duckplayer/impl/DuckPlayerScriptsJsMessaging.kt [hunk 3]
```diff
@@ -119,7 +120,9 @@ class DuckPlayerScriptsJsMessaging @Inject constructor(
             jsMessageCallback?.process(featureName, jsMessage.method, jsMessage.id ?: "", jsMessage.params)
         }
 
-        override val allowedDomains: List<String> = emptyList()
+        override val allowedDomains: List<String> = listOf(
+            runBlocking { duckPlayer.getYouTubeEmbedUrl() },
```
Comment (joshliebe) [inline duckplayer/duckplayer-impl/src/main/java/com/duckduckgo/duckplayer/impl/DuckPlayerScriptsJsMessaging.kt:124]: _Observation_: I've noticed that the duck player code uses `runBlocking` in a few places. While it functions as expected, could this increase the risk of ANRs when rolled out?

**Review by joshliebe — APPROVED** [review]
Tested and works as expected 👍 I just had the one question: https://github.com/duckduckgo/Android/pull/5053/files#r1778524580

**Pull request merged**